endpoint/endpoint.go

```diff
@@ -31,15 +31,18 @@ type Endpoint struct {
 	DNSName string
 	// The target the DNS record points to
 	Target string
+	// RecordType type of record, e.g. CNAME, A, TXT etc
+	RecordType string
 	// Labels stores labels defined for the Endpoint
 	Labels map[string]string
 }
 
 // NewEndpoint initialization method to be used to create an endpoint
-func NewEndpoint(dnsName, target string) *Endpoint {
+func NewEndpoint(dnsName, target, recordType string) *Endpoint {
 	return &Endpoint{
-		DNSName: strings.TrimSuffix(dnsName, "."),
-		Target:  strings.TrimSuffix(target, "."),
-		Labels:  map[string]string{},
+		DNSName:    strings.TrimSuffix(dnsName, "."),
+		Target:     strings.TrimSuffix(target, "."),
```

Comment:
I would strongly suggest to use a method for this: `sanitizeHostname` or `removeTrailingDot` as it makes it more obvious to the reader. However, not in this PR.